src/aws.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable camelcase */
+import AWS from 'aws-sdk'
+
+AWS.config.update({
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+  region: process.env.AWS_REGION || 'us-west-2',
+  apiVersions: {
+    lambda: '2015-03-31',
+    s3: '2006-03-01'
+  }
+})
+
+const bucketName = process.env.AWS_BUCKET || 'sinopia-marc-development'
+const lambdaName = process.env.AWS_RDF2MARC_LAMBDA || 'sinopia-rdf2marc-development'
+
+export const requestMarc = (resourceUri, resourceId, username, timestamp) => {
+  // These are tied to the user so that can list all of the MARC records requested by a user.
+  const marcKey = `${username}/${resourceId}/${timestamp}`
+  const marcPath = `${marcKey}/record.mar`
+  const marcTxtPath = `${marcKey}/record.txt`
+  const errorPath = `${marcKey}/error.txt`
+  // Invoke lambda.
+    const params = {
+      FunctionName: lambdaName,
+      InvokeArgs: JSON.stringify({
+      instance_uri: resourceUri,
+      marc_path: marcPath,
+      marc_txt_path: marcTxtPath,
+      error_path: errorPath,
+      bucket: bucketName
+    })
+  }
+  const lambda = new AWS.Lambda()
+  return new Promise((resolve, reject) => {
+    lambda.invokeAsync(params, (err) => {
+      if(err) reject(err)
+      else resolve()
+    })
+  })
+}
+
+export const hasMarc = (resourceId, username, timestamp) => {
+  const params = {
+    Bucket: bucketName,
+    Prefix: `marc/${username}/${resourceId}/${timestamp}`
```

Comment:
So we could have duplicate MARC records for the same BF Instance if two different users request conversion?

Reply:
Yes.